torch_geometric/nn/conv/message_passing.py

```diff
@@ -612,15 +613,52 @@ def jittable(self, typing: Optional[str] = None):
             prop_types = split_types_repr(match.group(1))
             prop_types = dict([re.split(r'\s*:\s*', t) for t in prop_types])
 
+        # Find and parse `edge_updater` types to format `{arg1: type1, ...}`.
+        # First test whether forward uses `edge_updater`
+        # TODO: Replace with AST search to see if the function is used
+        edge_updater_used = re.search(r'.*edge_updater.*', source)
+        if edge_updater_used:
+            if hasattr(self, 'edge_updater_type'):
+                edge_updater_types = {
+                    k: sanitize(str(v))
+                    for k, v in self.edge_updater.items()
+                }
+            else:
+                match = re.search(r'#\s*edge_updater_types:\s*\((.*)\)',
+                                  source)
+                if match is None:
+                    raise TypeError(
+                        'TorchScript support requires the definition of the '
+                        'types passed to `edge_updater()`. Please specificy '
+                        'them via\n\n edge_updater_types = {"arg1": type1, '
+                        '"arg2": type2, ... }\n\n or via\n\n'
+                        '# edge_updater_types: (arg1: type1, arg2: type2, ...)'
+                        '\n\ninside the `MessagePassing` module.')
+                edge_updater_types = split_types_repr(match.group(1))
+                edge_updater_types = dict(
+                    [re.split(r'\s*:\s*', t) for t in edge_updater_types])
+        else:
+            # dummy to be used
+            edge_updater_types = {'x': 'Tensor'}
```

Comment:
This is a bit nasty.. but right now without this it generates an invalid function signature. Not sure if this needs to be cleaned now or this works okay for you?

Reply:
Do you know why it needs it? What's the error message when we don't do that?

Reply:
I looked at this again and the main reason this is causing a problem is the way jinja the templating works now in my function
```def edge_updater(self, edge_index, {{ edge_updater_types.keys()|join(', ') }}):```
this is simple to fix though, I'll push a change

Reply:
Super!